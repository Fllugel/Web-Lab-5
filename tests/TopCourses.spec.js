import { shallowMount } from '@vue/test-utils';
import TopCourses from '@/components/MainContent.vue';  // Adjust the import path as needed

describe('TopCourses.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(TopCourses, {
      data() {
        return {
          courses: [
            { id: 1, image: 'image1.jpg', start_date: '2023-01-01', end_date: '2023-01-31' },
            { id: 2, image: 'image2.jpg', start_date: '2023-02-01', end_date: '2023-02-28' },
            { id: 9, image: 'image9.jpg', start_date: '2023-03-01', end_date: '2023-03-31' },
          ],
          coursesLoaded: true,
        };
      }
    });
  });

  it('renders the component', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('displays the correct number of courses', () => {
    const courseElements = wrapper.findAll('.ui.centered.card');
    expect(courseElements.length).toBe(3);
  });

  it('displays course details correctly', () => {
    const firstCourse = wrapper.findAll('.ui.centered.card').at(0);
    expect(firstCourse.find('img.image-card').attributes('src')).toBe('image1.jpg');
    expect(firstCourse.find('.header').text()).toBe('');
    expect(firstCourse.find('.meta').text()).toContain('Start Date: 2023-01-01');
    expect(firstCourse.find('.meta').text()).toContain('End Date: 2023-01-31');
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
