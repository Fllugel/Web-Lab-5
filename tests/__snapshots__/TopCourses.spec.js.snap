// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopCourses.vue matches the snapshot 1`] = `
"<div class="ui vertical container">
  <h1 class="ui horizontal divider title-home">TOP COURSES</h1>
  <div class="ui three column doubling grid">
    <!-- Display courses using Vue data -->
    <div class="column">
      <div class="ui special cards">
        <div class="ui centered card">
          <div class="blurring dimmable image"><img class="image-card" src="image1.jpg" alt="Course Image"></div>
          <div class="content"><a class="header"></a>
            <div class="meta"><span class="date">Start Date: 2023-01-01</span><br><span class="date">End Date: 2023-01-31</span></div>
          </div>
          <div class="ui green button fluid"><i class="cart icon"></i> Buy now </div>
        </div>
      </div>
    </div>
    <div class="column">
      <div class="ui special cards">
        <div class="ui centered card">
          <div class="blurring dimmable image"><img class="image-card" src="image2.jpg" alt="Course Image"></div>
          <div class="content"><a class="header"></a>
            <div class="meta"><span class="date">Start Date: 2023-02-01</span><br><span class="date">End Date: 2023-02-28</span></div>
          </div>
          <div class="ui green button fluid"><i class="cart icon"></i> Buy now </div>
        </div>
      </div>
    </div>
    <div class="column">
      <div class="ui special cards">
        <div class="ui centered card">
          <div class="blurring dimmable image"><img class="image-card" src="image9.jpg" alt="Course Image"></div>
          <div class="content"><a class="header"></a>
            <div class="meta"><span class="date">Start Date: 2023-03-01</span><br><span class="date">End Date: 2023-03-31</span></div>
          </div>
          <div class="ui green button fluid"><i class="cart icon"></i> Buy now </div>
        </div>
      </div>
    </div>
  </div>
  <div class="ui section divider"></div>
</div>"
`;
